src/prefect/core/task.py

```diff
@@ -368,6 +373,28 @@ def __init__(
             raise ValueError(
                 "A datetime.timedelta `retry_delay` must be provided if max_retries > 0"
             )
+        if retry_on and not max_retries > 0:
+            raise ValueError(
+                "A numer of `max_retries` must be provided if `retry_on` is set."
```

Comment:
Number

Reply:
👍 
Do these as suggestions so we can just fix them immediately
```suggestion
                "A number of `max_retries` must be provided if `retry_on` is set."
```